src/logic/equiv/fin.lean

```diff
@@ -424,6 +425,40 @@ def fin_prod_fin_equiv : fin m × fin n ≃ fin (m * n) :=
         ... = y.1 : nat.mod_eq_of_lt y.2),
   right_inv := λ x, fin.eq_of_veq $ nat.mod_add_div _ _ }
 
+/-- The equivalence between `a` and `(a / n, a % n)` for nonzero `n`.
+
+This is like `fin_prod_fin_equiv.symm` but with `m` infinite.
+See `nat.div_mod_unique` for a similar propositional statement. -/
+@[simps]
+def nat.div_mod_equiv (n : ℕ) [ne_zero n] : ℕ ≃ ℕ × fin n :=
+{ to_fun := λ a, (a / n, ↑a),
+  inv_fun := λ p, p.1 * n + ↑p.2,  -- TODO: is there a canonical order of `*` and `+` here?
+  left_inv := λ a, nat.div_add_mod' _ _,
+  right_inv := λ p, begin
+    refine prod.ext _ (fin.ext $ nat.mul_add_mod_of_lt p.2.is_lt),
+    dsimp only,
+    rw [add_comm, nat.add_mul_div_right _ _ (ne_zero.pos n), nat.div_eq_of_lt p.2.is_lt, zero_add],
+  end }
+
+/-- The equivalence between `a` and `(a / n, a % n)` for nonzero `n`.
```

Comment:
I interpret this sentence as saying that `a` and `(a / n, a % n)` are types you're writing an equivalence between.
```suggestion
/-- The equivalence between `ℤ` and `ℤ × fin n` induced by `a ↦ (a / n, a % n)` for nonzero `n`.
```
or something like that

Reply:
```suggestion
/-- The equivalence induced by `a ↦ (a / n, a % n)` for nonzero `n`.
```